docs/src/lib/indexing.md

```diff
@@ -197,3 +197,26 @@ Note that `sdf[!, col] .= v` and `sdf[!, cols] .= v` syntaxes are not allowed as
 
 If column indexing using `Symbol` names in `cols` is performed, the order of columns in the operation is specified
 by the order of names.
+
+
+## Indexing `GroupedDataFrame`s
+
+A [`GroupedDataFrame`](@ref) can behave as either an `AbstractVector` or `AbstractDict` depending on the type
+of index used. Integers (or arrays of them) trigger vector-like indexing while `Tuples`s and `NamedTuple`s
+trigger dictionary-like indexing. An intermediate between the two is the [`GroupKey`](@ref) type
+returned by [`keys(::GroupedDataFrame)`](@ref), which behaves similarly to a `NamedTuple` but has
+performance on par with integer indexing.
+
+The "elements" or "values" of a `GroupedDataFrame` are [`SubDataFrame`](@ref)s of its parent.
+
+* `gd[i::Integer]` -> Get the `i`th group.
+* `gd[a::AbstractArray{<:Integer}]` -> Get a reduced `GroupedDataFrame` containing only the groups
```

Comment:
This should be a correct contract - right? If yes - can we have all these options in the tests. Thank you!
```suggestion
* `gd[a::Union{AbstractVector{<:Integer}, Colon, Not}]` -> Get a reduced `GroupedDataFrame` containing only the groups
```

Reply:
This behavior hasn't changed in this PR (aside from restricting the signature), so I'd rather update this in a follow-up PR if that's alright.

Reply:
OK